### PR TITLE
add --debug option to setup.py

### PR DESCRIPTION
**Pull request by breznak**
Fixes: #1906

### CODE REVIEW

**Comment by breznak** [comment]
@rhyolight ok when :green_apple: ?

File: setup.py [hunk 1]
```diff
@@ -367,16 +373,21 @@ def getExtensionModules(nupicCoreReleaseDir, platform, bitness, cmdOptions=None)
...
+    if cmdOptions is not None and getCommandLineOption("optimizations-native", cmdOptions):
+      commonCompileFlags.append("-march=native")
+      commonCompileFlags.append("-O3")
+      commonLinkFlags.append("-O3")
+    if cmdOptions is not None and getCommandLineOption("optimizations-lto", cmdOptions):
+      commonCompileFlags.append("-fuse-linker-plugin")
+      commonCompileFlags.append("-flto-report")
+      commonCompileFlags.append("-fuse-ld=gold")
+      commonCompileFlags.append("-flto")
+      commonLinkFlags.append("-flto")
```
Comment (rhyolight) [inline setup.py]: I would make this simpler by updating `getCommandLineOption()` to handle `None`. Then you could just make it:

```
if getCommandLineOption("debug", cmdOptions):
    ...
else if getCommandLineOption("optimizations-native", cmdOptions):
    ...
else if getCommandLineOption("optimizations-lto", cmdOptions):
    ...
```

Comment (breznak) [inline setup.py]: nah, you don't want "optimizations-native" and "optimizations-lto" in
separate `else if` branches, that's preventing this nicer solution.

Comment (rhyolight) [inline setup.py]: Whoops, I meant:

```
if getCommandLineOption("debug", cmdOptions):
    ...
else:
    if getCommandLineOption("optimizations-native", cmdOptions):
        ...
    else if getCommandLineOption("optimizations-lto", cmdOptions):
        ...
```

This just prevents all the null-checks and makes it easier to read.

**Comment by breznak** [comment]
Thanks @rhyolight looks better now!

**Pull request merged**